lib/Facebook.js

```diff
@@ -129,31 +134,51 @@ function Facebookbot(configuration) {
             //Add Access Token to outgoing request
             facebook_message.access_token = configuration.access_token;
 
-            request({
-                method: 'POST',
-                json: true,
-                headers: {
-                    'content-type': 'application/json',
-                },
-                body: facebook_message,
-                uri: 'https://' + api_host + '/v2.6/me/messages'
-            },
-                function(err, res, body) {
+            var requestObj = {};
+            if ('filedata' in facebook_message) {
+              var prepareFormdata = {};
+              for (key in facebook_message) {
+                if (key !== 'filedata' && typeof facebook_message[key] === 'object') {
+                  prepareFormdata[key] = JSON.stringify(facebook_message[key]);
+                } else if (key === 'filedata') {
+                  if (!fs.existsSync(facebook_message[key])) return cb(new Error('filedata not exist'));
 
+                  prepareFormdata[key] = fs.createReadStream(facebook_message[key]);
+                } else {
+                  prepareFormdata[key] = facebook_message[key];
+                }
+              }
+              requestObj = {
+                  method: 'POST',
+                  formData: prepareFormdata,
+                  uri: 'https://' + api_host + '/v2.6/me/messages'
```

Comment:
Haven't used the upload api myself, but the [docs](https://developers.facebook.com/docs/messenger-platform/send-api-reference/attachment-upload) say the url for uploads is:
`v2.6/me/message_attachments`